src/main/java/net/dries007/tfc/CommonEventHandler.java

```diff
@@ -144,15 +167,32 @@ public static void onRightClickBlock(PlayerInteractEvent.RightClickBlock event)
         IPlaceableItem placeable = IPlaceableItem.Impl.getPlaceable(stack);
         if (placeable != null)
         {
-            if (placeable.placeItemInWorld(world, pos, stack, player, event.getFace(), event.getHitVec()))
+            // Dirty trick to fire onBlockActivated first
```

Comment:
I don't like the fact that this is half-implemented. It should follow the exact same logic chain that vanilla does in `PlayerInteractionManager#handleRightClick`. That is, always cancel the event if something matches, then duplicate the rest of the function (all of the relevant context is available). When we get to `onItemUse`, handle our placeable instead of `onItemUse`

Minecraft's right click handling is very detailed and handles a lot of cases. I do not think it's a good idea to just blindly fire `onBlockActivated` without properly considering the effects that may have